### PR TITLE
Clean up buffered signal test

### PR DESCRIPTION
**Pull request by dandavison**
Clean up a test, similar to https://github.com/temporalio/temporal/pull/5794. No functional changes.

### CODE REVIEW

File: tests/client_suite.go [hunk 1]
```diff
@@ -1256,38 +1247,32 @@ func (s *ClientFunctionalSuite) Test_BufferedSignalCausesUnhandledCommandAndSche
...
 	}
 
 	s.NotNil(workflowRun)
 	s.True(workflowRun.GetRunID() != "")
+	tv = tv.WithRunID(workflowRun.GetRunID())
 
 	// block until first workflow task started
-	select {
-	case <-sigReadyToSendChan:
-	case <-ctx.Done():
```
Comment (stephanos) [inline tests/client_suite.go:1287]: I _think_ by omitting `<-ctx.Done` (here and above) the test _might_ get stuck and run until the GHA terminates.

**Review by stephanos — APPROVED** [review]

File: tests/client_suite.go (hunk 1 above)
Comment (tdeebswihart) [reply tests/client_suite.go:1287]: Yep! This is a recipe for a deadlock

File: tests/client_suite.go [hunk 2]
```diff
@@ -1215,39 +1215,30 @@ func (s *ClientFunctionalSuite) Test_ActivityTimeouts() {
...
-		select {
-		case <-sigSendDoneChan:
-		case <-ctx.Done():
-		}
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	tv := testvars.New(s.T().Name()).WithTaskQueue(s.taskQueue)
 
+	sigReadyToSendChan := make(chan bool, 1)
+	sigSendDoneChan := make(chan bool)
```
[the remark this answers is not in the document]
Comment (dandavison) [reply tests/client_suite.go]: Thanks, I had initially been thinking that `bool` was better for readability reasons in the case where memory usage is not a consideration, but you made me think again and I realize that in this case I'm only ever going to write `true` to the channel, so writing 0 bits makes more sense than writing 1.

File: tests/client_suite.go (hunk 1 above)
Comment (dandavison) [reply tests/client_suite.go:1287]: Yep, reinstated, thanks. (The other `select` that I removed I believe is not necessary).

**Pull request merged**